From 08e9e23189305e00e1b7d0fa65b90708d86e60d0 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Thu, 13 Jan 2022 14:15:37 +0200
Subject: [PATCH] [SW-69642] thermal: fix calculation for 32bit

Usage of 'long' in arm is 64bit while in arc it is 32bit. In order to
make sure the thermal mC calculation is not performed differently
between the two platform a size specific type is used long->s64.

Change-Id: I173c4b85144184f1ce426016d313de3b32fa40f9
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/lib/preboot/thermal_v2.c b/lib/preboot/thermal_v2.c
index a247ea1..ebd4fed 100644
--- a/lib/preboot/thermal_v2.c
+++ b/lib/preboot/thermal_v2.c
@@ -142,7 +142,7 @@
 
 long hl_tmp_temp_reg_to_mC(int val, int ts)
 {
-	long tmp = (val & HL_TS_00_SDIF_DATA_SMPL_DATA_MASK);
+	s64 tmp = (val & HL_TS_00_SDIF_DATA_SMPL_DATA_MASK);
 	s32 coef_a_g, coef_b_h, coef_div;
 
 	coef_a_g = hl_ts_data[ts].hl_tmp_data.coef_a_g;
